canonical-web-and-design/vanilla-framework
# Ensure sidenav is only tabable when in view

## Done

- Added functionality using tabindex to ensure the side navigation is only tab-able when it's in view (aria-hidden can't be used on focusable elements)
- Ensure on page load depending on screen size the side is either tab-able or not 
- Update the tab index when the screen size changes and the drawer is hidden or in view
- Update tab index when the toggle side nav button is clicked 
- Added accessibility notes

Fixes #4134 

## QA

- Open [demo](https://vanilla-framework-4371.demos.haus/docs)
- First, load the page on a screen size bigger than 1035 when the side nav is visible by default
- Check you can tab through the links
- Then change the screen size so the side nav is hidden. Check when you tab through the page it ignores the side nav links
- Make the screen bigger again to reveal the side nav and check you can tab through the links
- Make it smaller again, and use the "Toggle side navigation" button to open the side nav drawer. Check you can tab through the links
- Close the side nav with the button, and then make sure the links return to being non tab-able. 
- When opening and closing toggle drawer button using the keyboard, make sure the focus lands on the "toggle side navigation" button inside and outside the drawer


### Check if PR is ready for release

If this PR contains Vanilla SCSS code changes, it should contain the following changes to make sure it's ready for the release:

  - `Feature 🎁`, `Breaking Change 💣`, `Bug 🐛`, `Documentation 📝`, `Maintenance 🔨`.
  - if CSS class names are not changed it can be bugfix relesase (x.x.**X**)
  - if CSS class names are changed/added/removed it should be minor version (x.**X**.0)
  - see the [wiki for more details](https://github.com/canonical-web-and-design/vanilla-framework/wiki/Release-process#pre-release-tasks)
